src/hermes/commands/deposit/base.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: 2023 Helmholtz-Zentrum Dresden-Rossendorf (HZDR)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileContributor: David Pape
+
+import click
+
+from hermes.model.context import CodeMetaContext
+
+
+# TODO: Abstract base class?
+class BaseDepositPlugin:
+    def __init__(self, click_ctx: click.Context, ctx: CodeMetaContext) -> None:
+        self.click_ctx = click_ctx
+        self.ctx = ctx
+
+    def __call__(self) -> None:
+        # TODO: Decide here which of initial/new/... to run?
+        steps = [
+            "prepare",
+            "map",
+            "create_initial_version",
+            "create_new_version",
+            "update_metadata",
+            "delete_artifacts",
+            "upload_artifacts",
+            "publish",
+        ]
+
+        for step in steps:
+            getattr(self, step)()
```

Comment:
**issue:** It does not feel right to handle it this way... lots of unintended stuff could happen here, some magic might not work. Also all the functions are state-dependant which in my eyes is not the best solution here.